core/src/test/java/io/crate/monitor/SysInfoUtilTest.java

```diff
@@ -117,6 +121,40 @@ public void testParseRedHatVendor() throws IOException {
         assertThat(util.info().vendorCodeName(), is("Santiago"));
     }
 
+    @Test
+    public void testGenericVendor() throws IOException {
+        SysInfoUtil util = genericSysInfoUtil();
```

Comment:
btw, based on the name of the class I wouldn't have expected it do contain state (which it only does indirectly by keeping the reference to `SysInfo`.  Having a `SysInfo.Builder` instead, which contains the parse methods would imho been clearer.

Reply:
I named it SysInfoUtil because besides of providing the system information via `sysInfo()` it also has a `getSystemUptime()` method.
In general I agree that the collecting of the system information could be encapsulated in a builder, so actually it's the other way around than it is now.